### DISCUSSION TITLE
Policy for Resource?

### DISCUSSION THREAD

**Discussion by robertpotkowa** [filamentphp/filament#7928] [Help]
I have to Resources, UserResource, EmployeeResource. Both access on the User model. But how is it possible to policy both independently, when both Resources use the UserPolicy from the User::model? Is it possible maybe to set, which Policy the Resource should use? Or can i only achive this, by extend the userModel with EmplyeeModel and create for that an own Policy?

**Comment by danharrin** [comment]
You could extend your model class and register different policies like you said, IMO that would be very clean. Or, you could override the authorize() / can() methods on the resource class to check a different policy. Or, you could just go ahead and use the same policy for both resources and check the type of the user inside it. There are many different approaches here.

**Comment by robertpotkowa** [comment]
Ok, but when i use the policy for both resources, how can i check inside the policy which resource already usese the policy to switch between users.

**Reply by danharrin** [reply]
Can't you just check the type of the user in the policy based on some attribute?

**Comment by robertpotkowa** [comment]
Yes i can. But when i use the same user Policy for both Resources: UserResource and EmplyeeResource, how can the policy know, what type of Resource it is? 
UserResource -> User::model, (Show only $user->role == 'admin')
EmployeeResource -> User::model (Show $user->role == 'employee' OR $user->role == 'admin')

How the Policy knows, what type of Resource it?

**Reply by danharrin** [reply]
Because in the policy you check `$user->role`?

**Reply by michaelklopf** [reply]
He probably meant when listing records. I had the same problem right now. Have a user resource that should be viewed only by the person with owner permissions.

The names of the functions to be overwritten are in `Filament\Resources\Resource.php` like 

```
    public static function canViewAny(): bool
    {
        return true;
    }
```

**Reply by keatliang2005** [reply]
I came across this issue as well, a closer look into the documentation, it's intended for page, but it works on resource
https://filamentphp.com/docs/3.x/panels/pages#authorization

Method below will have a blanket check, and load specific user type

User resource
```php
  public static function canAccess(): bool
  {
     $user = auth()->user();
     return $user->role == 'admin';
  }
```

EmployeeResource
```php
  public static function canAccess(): bool
  {
     $user = auth()->user();
     return $user->role == 'employee' || $user->role == 'admin';
  }

  public static function getEloquentQuery(): Builder
  {
      return parent::getEloquentQuery()->::whereRelation('roles', 'name', 'employee');
  }
```

but i am not sure how secure it is, example directly hit liveware endpoint to create/update methods on these resources
